AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientTests.swift

```diff
@@ -793,6 +795,37 @@ class AWSMobileClientTests: XCTestCase {
         
     }
     
+    func testGetIdentityId() {
+        XCTAssertNil(AWSMobileClient.sharedInstance().identityId, "Identity Id should be nil after initialize.")
+
+        let identityIdExpectation = expectation(description: "IdentityID should be retrieved successfully")
+        AWSMobileClient.sharedInstance().getIdentityId().continueWith(block: { (task) -> Any? in
+            XCTAssertNil(task.error)
+            XCTAssertNotNil(task.result, "getIdentityId should not return nil.")
+            identityIdExpectation.fulfill()
+            return nil
+        })
+        wait(for: [identityIdExpectation], timeout: 5)
+
+        guard AWSMobileClient.sharedInstance().identityId != nil else {
+            XCTFail("Could not find identityId.")
+            return
+        }
+
+        let verifyCredentialsExpectation = expectation(description: "Credentials should be retrieved successfully")
```

Comment:
I'm losing track of what you're actually testing in this. Are you asserting in the second half that `getAWSCredentials` doesn't incorrectly _clear_ the identity ID? I think this would be clearer with a separate test. In general let's try to minimize the number of assertions in each of our tests, with 1 being the ideal number.

Reply:
Removed `getAWSCredentials` from the test.